src/NServiceBus.Core/Routing/MessageDrivenSubscriptions/MessageDrivenSubscriptions.cs

```diff
@@ -90,6 +91,26 @@ protected internal override void Setup(FeatureConfigurationContext context)
                 context.Pipeline.Register(new SendOnlySubscribeTerminator(), "Throws an exception when trying to subscribe from a send-only endpoint");
                 context.Pipeline.Register(new SendOnlyUnsubscribeTerminator(), "Throws an exception when trying to unsubscribe from a send-only endpoint");
             }
+
+            context.Container.ConfigureComponent<CallInit>(DependencyLifecycle.SingleInstance);
+
+            context.RegisterStartupTask(b => b.Build<CallInit>());
```

Comment:
should we also do this here: https://github.com/Particular/NServiceBus/blob/master/src/NServiceBus.Core/Routing/SubscriptionMigrationMode/SubscriptionMigrationMode.cs ?

Reply:
Done